src/index.ts

```diff
@@ -91,6 +93,16 @@ async function metadataAccessor<T>(
   }
 }
 
+async function fastFailMetadataRequest<T>(
+  options: GaxiosOptions
+): Promise<GaxiosResponse> {
+  const secondaryOptions = {
+    ...options,
+    url: options.url!.replace(BASE_URL, SECONDARY_BASE_URL),
+  };
+  return Promise.race([request<T>(options), request<T>(secondaryOptions)]);
```

Comment:
Ideally the race should happen only for the first request. Once we have established success once, we should switch to using the primary (i.e. IP) mode of connection from then onwards.

Reply:
Maybe this is easier to do if you went back to your original application of the race only on `isAvailable`.

Reply:
I agree, I think `isAvailable` is the main culprit right now causing user-facing issues, let's address this bug first.

Reply:
:+1: I think this is a great start, and I'll test thoroughly with logging.